### PR TITLE
Backlog 15687 hide types

### PR DESCRIPTION
**Pull request by AKarmanov**
## JIRA

https://jira.jahia.org/browse/BACKLOG-15687

## Description

Migration script and a short readme update for the migration part.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
 Sitemap (optional module)
 =========================
 
+## Migration
+
+On Jahia >= 8.0.3 migration should happen automatically provided that you have previous version of the module installed.
```
Comment (rknj) [inline README.md:6]: Automatically only if the groovy script is added in the Jahia patches folder

Comment (AKarmanov) [reply README.md:6]: It is in the patches folder by default

Comment (rknj) [reply README.md:6]: I meant in the Jahia patches folder. sitemap is not provided with the Jahia distribution where the patches are now executed at startup. But if the user add the new version in the migrated Jahia, will the patch provided by the module be executed at startup like the others ?

Comment (AKarmanov) [reply README.md:6]: Yes, the patches in the module are independent of what we add in Jahia so they will run on first startup of this version on 8.0.3 Jahia.

**Review by gflores-jahia — APPROVED** [review]
ProcessJntPage and ProcessJntContent could've been implemented as a single class and just parameterize the type on create to avoid code duplication. But not as important for a one-time clean up script. Something to consider if we ever make any changes to it again.

**Pull request merged**